dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/storage/RValueDb.scala

```diff
@@ -0,0 +1,35 @@
+package org.bitcoins.dlc.oracle.storage
+
+import org.bitcoins.core.hd._
+import org.bitcoins.crypto.SchnorrNonce
+
+case class RValueDb(
+    nonce: SchnorrNonce,
+    label: String,
+    purpose: HDPurpose,
+    accountCoin: HDCoinType,
+    accountIndex: Int,
+    chainType: Int,
+    keyIndex: Int) {
+
```

Comment:
Shouldn't there be an invariant here saying that it `keyIndex` is hardened?

Reply:
It is always hardened here

Reply:
Then why not add the invariant? They we know for sure it's never hardened and can't be misused by someone accidentally in the future

Reply:
There is no invariant to add, `Int` is neither hardened nor unhardened. We can only tell with a `UInt32`

Reply:
Is it because `keyIndex` is relative to the `ExtKey.hardenedIdx` ?

Reply:
```
val hardenedIdx = UInt32(NumberUtil.pow2(31).toLong)

case class BIP32Node(index: Int, hardened: Boolean) {
  require(index >= 0, s"BIP32 node index must be positive! Got $index")

  /**
    * Converts this node to a BIP32 notation
    * unsigned 32 bit integer
    */
  def toUInt32: UInt32 =
    if (hardened) ExtKey.hardenedIdx + UInt32(index.toLong)
    else UInt32(index)
}
```